kazoo/protocol/connection.py

```diff
@@ -439,7 +439,7 @@ def _send_request(self, read_timeout, connect_timeout):
         if request.type == Auth.type:
             xid = AUTH_XID
         else:
-            self._xid += 1
+            self._xid = (self._xid % 2147483647) + 1
```

Comment:
Can u add a comment as to why u picked this number; and a link ideally to the zookeeper docs so that if it ever changes we can know. Doesn't it also rollover to -2147483648 instead of zero (I though this was a signed int in java)